### PR TITLE
UIP-16 September dependency update

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

Update dependencies
* Remove `requirements-no-deps.txt` as it's not necessary anymore.
* Remove various scientific programming requirements from the narrative repo, they're now in the base image
* Remove other non-required requirements (other helpful Python things that the narrative doesn't directly need) as they're now in the base image.
* Remove semantic_version as it's only used in one place, and not effectively
* Update remaining dependencies to the latest version
* Update npm dependencies
* Bump Jupyter notebook and IPython versions

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/UIP-16
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [ ] Tests pass locally and in GitHub Actions
- [ ] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [x] (Python) I have run Black and Flake8 on changed Python code manually or with a git precommit hook
- [x] Any dependent changes have been merged and published in downstream modules

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: src/requirements.txt [hunk 1]
```diff
@@ -1,27 +1,21 @@
...
 pyopenssl==22.0.0
 pytest==7.0.1
 pytest-cov==3.0.0
-requests==2.27.1
+requests==2.28.1
 rsa==4.9
-semantic_version==2.10.0
-sympy==1.9
 terminado==0.12.1
 # ensure pip installs all deps
```
Comment (ialarmedalien) [inline src/requirements.txt:21]: looks like this line can be removed now

File: src/requirements.txt [hunk 2]
```diff
@@ -1,27 +1,21 @@
 beautifulsoup4==4.11.1
-black>=20.8b1
-bokeh==2.3.3
-chart_studio==1.1.0
-coverage==6.2
-cryptography==37.0.4
+black==22.8.0
```
Comment (ialarmedalien) [inline src/requirements.txt:2]: it's probably worth putting a comment at the top of this file noting that other python modules are added by the installer script.

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -2,7 +2,26 @@
...
+  - jinja2 3.0.3 -> 3.1.2
+  - jupyter-console 6.4.3 -> 6.4.4
+  - requests 2.27.1 -> 2.28.1
+
+Removed dependencies  
+**note** a number of these were removed from the narrative, but left in the narrative base image. The `requirements.txt` in this repo is intended for those packages required for the narrative to run. Any other packages, including scientific programming ones, useful for either KBase Apps or other manual use in Narrative code cells are included in the narrative base image (see [narrative-base-image](https://github.com/kbase/narrative-base-image) for more details). Also note that most of these are duplicated and included in that image - these are still necessary for narrative functionality when installed locally (i.e. not in a Docker image), so they're left in.
+  - plotly - not necessary for core narrative function, moved to narrative base image
+  - semantic_version -- removed, wasn't used effectively, and removed the need for it
+  - sklearn - still included, installed manually alongside clustergrammer in the install script, so the requirements.txt was removed
+  - sympy - not necessary for core narrative funciton, moved to narrative base image
```
Comment (ialarmedalien) [inline RELEASE_NOTES.md:24]:
```
s/funciton/function/
```

**Review by ialarmedalien — APPROVED** [review]

**Comment by briehl** [comment]
@ialarmedalien I'm gonna hold off until the base-image update and release, then test with that one and merge.

**Pull request merged**